common4j/src/main/com/microsoft/identity/common/java/challengehandlers/PKeyAuthChallengeFactory.java

```diff
@@ -59,66 +58,69 @@
      * This parses the redirectURI for challenge components and produces
      * response object.
      *
+     * This is retrieved from response from auth endpoint
+     * (read: it would be triggered in interactive flow only).
+     *
      * @param redirectUri Location: urn:http-auth:CertAuth?Nonce=<noncevalue>
      *                    &CertAuthorities=<distinguished names of CAs>&Version=1.0
      *                    &SubmitUrl=<URL to submit response>&Context=<server state that
      *                    client must convey back>
      * @return Return PKeyAuth challenge object
      */
-    public PKeyAuthChallenge getPKeyAuthChallenge(@NonNull final String redirectUri) throws ClientException {
+    public PKeyAuthChallenge getPKeyAuthChallengeFromWebViewRedirect(@NonNull final String redirectUri) throws ClientException {
         //get the PKeyAuthChallenge from redirect Uri sent from authorization endpoint
         final Map<String, String> parameters = UrlUtil.getParameters(redirectUri);
-        validatePKeyAuthChallenge(parameters);
+        validatePKeyAuthChallengeFromWebViewRedirect(parameters);
+
+        final PKeyAuthChallenge.PKeyAuthChallengeBuilder builder = new PKeyAuthChallenge.PKeyAuthChallengeBuilder();
+        builder.nonce(parameters.get(Nonce.name().toLowerCase(Locale.US)))
+                .context(parameters.get(Context.name()))
+                .version(parameters.get(Version.name()))
+                .submitUrl(parameters.get(SubmitUrl.name()));
 
-        final PKeyAuthChallenge.Builder builder = new PKeyAuthChallenge.Builder();
-        builder.setNonce(parameters.get(Nonce.name().toLowerCase(Locale.US)))
-                .setContext(parameters.get(Context.name()))
-                .setCertAuthorities(StringUtil.getStringTokens(
-                        parameters.get(CertAuthorities.name()), CHALLENGE_REQUEST_CERT_AUTH_DELIMITER))
-                .setVersion(parameters.get(Version.name()))
-                .setSubmitUrl(parameters.get(SubmitUrl.name()));
+        if (parameters.containsKey(CertAuthorities.name())) {
+            String authorities = parameters.get(CertAuthorities.name());
```

Comment:
should this be final? the scope is small tho.

Reply:
doesn't hurt :D